utils/openshift_resource.py

```diff
@@ -30,12 +32,28 @@ def has_qontract_annotations(self):
             annotations = self.body['metadata']['annotations']
 
             assert annotations['qontract.integration'] == self.integration
-            assert annotations['qontract.integration_version'] is not None
+
+            integration_version = annotations['qontract.integration_version']
+
+            # handle old version that were not using semver
```

Comment:
Can we instead re-annotate all existing objects and add support only for semver instead of hardcoding this? I am not against this approach, just trying to understand in what use cases we'd need to run this code once the objects are using a new version.

Reply:
well, this saves up from re-annotate all existing resources, but it's indeed an option. I believe this should work, but your proposal also makes sense. wdyt @maorfr ?

Reply:
This is only for resources that were migrated before this PR is merged.

Reply:
I think we should re-annotate all resources now, rather then having some old ones floating around.